src/sys/getrandom.rs

```diff
@@ -0,0 +1,48 @@
+//! Wrapper around getrandom.
+
+use crate::Result;
+use libc;
+
+libc_enum! {
+    /// How the random bytes should be filled
+    #[repr(u32)]
+    #[non_exhaustive]
+    pub enum RandomMode{
+        /// If  this  bit is set, then random bytes are drawn from the
+        /// random source (i.e., the same source as the /dev/random device)
+        ///  instead of the urandom source.
+        GRND_RANDOM,
+        /// By default, when reading from the random source, getrandom()
+        /// blocks if no random bytes are available, and when reading
+        /// from the urandom source, it blocks if the entropy pool has
+        /// not yet been initialized. If the GRND_NONBLOCK flag is
+        /// set, then getrandom() does not block in these cases
+        GRND_NONBLOCK,
+    }
+}
+
+/// Returns a vector of random bytes
+pub fn getrandom(size: usize, mode: RandomMode) -> Result<Vec<u8>> {
+    let mut buffer = vec![0; size];
+    unsafe {
+        assert_eq!(
```

Comment:
The Linux man page lists a few ways that `getrandom` can fail - we should propagate these up via the `Result`, rather than asserting.

We also will want to expose the number of bytes actually read, since `GRND_NONBLOCK` may cause this not to write all of them.

Reply:
Perhaps the signature should be something like

```rust
pub fn getrandom(buffer: &mut [u8], mode: RandomMode) -> Result<usize> 
```

where the result indicates the number of bytes written?